stdlib/asm/math/secp256k1_base_field.masm

```diff
@@ -0,0 +1,547 @@
+#!
+#! See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L57-L76
+export.add
+  movupw.2
+
+  push.0
+  movup.5
+  u32overflowing_add3
+
+  movup.2
+  movup.5
+  u32overflowing_add3
+
+  movup.3
+  movup.5
+  u32overflowing_add3
+
+  movup.4
+  movup.5
+  u32overflowing_add3
+
+  movup.5
+  movup.9
+  u32overflowing_add3
+
+  movup.6
+  movup.9
+  u32overflowing_add3
+
+  movup.7
+  movup.9
+  u32overflowing_add3
+
+  movup.8
+  movup.9
+  u32overflowing_add3
+
+  movup.8
+  dup.1
+  push.977
+  u32overflowing_madd
+  drop
+
+  swap
+  movup.8
+  add
+
+  movup.2
+  movup.3
+  movup.4
+  movup.5
+  movup.6
+  movup.7
+  movup.6
+  movup.7
+end
+
+#! Given a secp256k1 base field element ( say a ) on stack, represented in Montgomery form 
+#! ( i.e. number having eight 32 -bit limbs ), following function negates it to
+#! field element a' | a' + a = 0
+#!
+#! Stack expected as below, holding input
+#!
+#! [a0, a1, a2, a3, a4, a5, a6, a7, ...] | a[0..8] is a secp256k1 base field element
+#!
+#! After finishing execution of this function, stack should look like
+#!
+#! [c0, c1, c2, c3, c4, c5, c6, c7, ...] | c[0..8] is a secp256k1 base field element
+#!
+#! See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L78-L96
+export.neg
+  push.0
+  swap
+  push.4294966319
+  exec.sbb
+
+  movup.2
+  push.4294967294
+  exec.sbb
+
+  movup.3
+  push.4294967295
+  exec.sbb
+
+  movup.4
+  push.4294967295
+  exec.sbb
+
+  movup.5
+  push.4294967295
+  exec.sbb
+
+  movup.6
+  push.4294967295
+  exec.sbb
+
+  movup.7
+  push.4294967295
+  exec.sbb
+
+  movup.8
+  push.4294967295
+  exec.sbb
+
+  drop
+  
+  swap
+  movup.2
+  movup.3
+  movup.4
+  movup.5
+  movup.6
+  movup.7
+end
+
+#! Given two secp256k1 base field elements, say a, b, ( represented in Montgomery form,
+#! each number having eight 32 -bit limbs ) on stack, following function computes modular
+#! subtraction of those two operands c = a + (-b) = a - b
+#!
+#! Stack expected as below, holding input
+#!
+#! [a0, a1, a2, a3, a4, a5, a6, a7, b0, b1, b2, b3, b4, b5, b6, b7, ...] | a[0..8], b[0..8] are secp256k1 base field elements
+#!
+#! After finishing execution of this function, stack should look like
+#!
+#! [c0, c1, c2, c3, c4, c5, c6, c7, ...] | c[0..8] is a secp256k1 base field element
+#!
+#! See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L98-L102
+export.sub
+  movupw.3
+  movupw.3
+
+  exec.neg
+  exec.add
+end
+
+#! Given a 256 -bit number on stack, represented in radix-2^32 form i.e. eight 32 -bit limbs, 
+#! this routine computes Montgomery representation of provided radix-2^32 number.
+#!
+#! Stack expected in form
+#!
+#!  [a0, a1, a2, a3, a4, a5, a6, a7, ...]
+#!
+#! Final stack should look like
+#!
+#! [a0`, a1`, a2`, a3`, a4`, a5`, a6`, a7`, ...]
```

Comment:
as noted in the docs, this does not format correctly. (and same for the `from_mont` doc comment below)

Reply:
See https://github.com/maticnetwork/miden/pull/481#discussion_r1036019553